src/main/java/com/cloudbees/jenkins/support/SupportAction.java

```diff
@@ -185,6 +181,78 @@ public void doDeleteBundles(StaplerRequest req, StaplerResponse rsp) throws Serv
         rsp.sendRedirect("");
     }
 
+    @RequirePOST
+    public void doDownloadBundles(StaplerRequest req, StaplerResponse rsp) throws ServletException, IOException {
+        JSONObject json = req.getSubmittedForm();
+        if (!json.has("bundles")) {
+            rsp.sendError(HttpServletResponse.SC_BAD_REQUEST);
+            return;
+        }
+
+        Set<String> bundlesToDownload = getSelectedBundles(req, json);
+        File fileToDownload = null;
+        if (bundlesToDownload.size() > 1) {
+            // more than one bundles were selected, create a zip file
+            fileToDownload = createZipFile(bundlesToDownload);
+        } else {
+            fileToDownload = new File(SupportPlugin.getRootDirectory(), bundlesToDownload.iterator().next());
+        }
+        logger.fine("Trying to download file "+ fileToDownload.getAbsolutePath());
+        try {
+            rsp.setContentType("application/zip");
+            rsp.addHeader("Content-Disposition", "inline; filename=" + fileToDownload.getName() + ";");
+            FileUtils.copyFile(fileToDownload, rsp.getOutputStream());
+        } catch (RuntimeException e) {
+            logger.log(Level.SEVERE, "Unable to download file " + fileToDownload.getAbsolutePath(), e);
+        } finally {
+            if (bundlesToDownload.size() > 1 && fileToDownload.delete()) {
+                logger.log(Level.FINE, "Temporary multiBundle file deleted: " + fileToDownload.getAbsolutePath());
+            } else {
+                logger.log(Level.SEVERE, "Unable to delete temporary multiBundle file: " + fileToDownload.getAbsolutePath());
+            }
+        }
+    }
+
+
+    private Set<String> getSelectedBundles(StaplerRequest req, JSONObject json) throws ServletException, IOException {
+        Set<String> bundles = new HashSet<>();
+        List<String> existingBundles = getBundles();
+        for (Selection s : req.bindJSONToList(Selection.class, json.get("bundles"))) {
+            if (s.isSelected()) {
+                if (existingBundles.contains(s.getName())) {
+                    bundles.add(s.getName());
+                } else {
+                    logger.log(Level.FINE, "The bundle selected {0} does not exist, so it will not be processed", s.getName());
+                }
+            }
+        }
+        return bundles;
+    }
+
+    private File createZipFile(Set<String> bundles) throws IOException {
+        File rootDirectory = SupportPlugin.getRootDirectory();
+        File zipFile = File.createTempFile(
```

Comment:
I would also mark the deletion of the temp file when Jenkins exits with [zipFile.deleteOnExit()](https://docs.oracle.com/javase/8/docs/api/java/io/File.html#deleteOnExit--).

Reply:
Done